staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -667,3 +674,82 @@ func forceUmount(path string) error {
 	}
 	return nil
 }
+
+// isMountFastPath is a method of detecting a mount that really fast.
+// It should return a true and no-error ONLY when it is an actual mount point.
+// In cases, where it is not guaranteed or cannot be determined, it should return an error.
+// For Linux, it avoids parsing /proc/mounts which is a very slow operation.
+func isMountFastPath(path string) (bool, error) {
+	// the second parameter should be nil always in production code path.
+	// it is used to inject mock function for openat2 for unit-testing.
+	return isMountPointUsingOpenAt2(path, nil)
+}
+
+// isMountPointUsingOpenAt2 uses openat2 syscall is a method of detecting a mount that works for all kinds
+// of mounts (incl. bind mounts), but requires a recent (v5.6+) linux kernel.
+// The code takes inspiration from here:
+// https://github.com/moby/sys/blob/40883be4345cf291513e2a4112896d393455fdfb/mountinfo/mounted_linux.go#L10
+// It expects that all symlinks have been evaluated.
+// If an ENOENT is returned by openat2, return os.ErrNotExist to satisfy isNotMountPoint return values.
+// openat2 is injected for testing purposes. In production code path, it should always be nil.
+func isMountPointUsingOpenAt2(path string, openat2 func(int, string, *unix.OpenHow) (int, error)) (bool, error) {
+
+	if openat2 == nil {
+		openat2 = unix.Openat2
+	}
+	// explicitly clean the path to remove trailing slashes.
+	// ref: https://pkg.go.dev/path/filepath#Clean
+	// otherwise, this will remove false +-ves or return errors.
+	// Ref: https://github.com/kubernetes/kubernetes/pull/105833#issuecomment-951092014
+	path = filepath.Clean(path)
+
+	// only work on absolute paths i.e starting with /
```

Comment:
I don't understand WHY this is important.  If it really is, can you comment on it here?

Reply:
This is important because we can cross different mount points and get either a false +ve or -ve if the path does not start with `/`;  I will add a comment as well with appropriate example.

Reply:
The comment still does not make much sense. It should say that it needs absolute path, I can read `if` below, it should says why it's important to have absolute paths.